Scripts/Commands: command .modify power (all powers)

Hadifaren opened the pull request:
`.modify power` `#powercount`

Closes #19288

**Target branch(es):** 3.3.5/master

- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)

Tested And Works

Hadifaren committed:
Merge  TrinityCore/master

Updated to latest

Hadifaren committed:
New Command for Modifying Various Powers

Hadifaren committed:
New Command for Modifying Various Powers

Hadifaren committed:
New Command for Modifying Various Powers

Hadifaren committed:
New Command for Modifying Various Powers

Hadifaren committed:
New Command for Modifying Various Powers

Hadifaren committed:
space

Hadifaren committed:
Update RBAC.h

Commit:
Update and rename xxxx_xx_xx_xx_auth.sql to 2017_mm_nn_xx_auth.sql

- make the SQL values match the scripted command values

Commit:
Update and rename xxxx_xx_xx_xx_world.sql to 2017_mm_dd_xx_world.sql

- use correct permission value of the scripted command
- codestyle corrections / improvements

Commit:
Rename 2017_mm_nn_xx_auth.sql to 2017_11_16_20833_auth.sql

- make the auth sql filename more relevant to this PR

tkrokli commented:
```
- "sql/updates/auth/2017_11_16_20833_auth.sql" is missing in the 'auth'.'updates' table.
error:
  The Database Updater is broken for the 'auth' database,
  because the applied update is missing in the 'auth'.'updates' table.
How to fix:
  Insert the missing names of the already applied sql updates
  to the 'updates' table of the 'auth' base dump ('sql/base/auth_database.sql').
```

Hadifaren committed:
Pain And Fury Languages

Hadifaren committed:
Update 2017_mm_dd_xx_world.sql

Hadifaren commented:
i dont know how to solve the error you writed ^_^
+ cannot understand how to fix

Hadifaren commented:
:| why its closed ! @tkrokli

tkrokli commented on src/server/scripts/Commands/cs_modify.cpp:
```diff
@@ -997,6 +998,66 @@ class modify_commandscript : public CommandScript
+		int8 const powerMultiplier = 10;
+		if (CheckModifyResources(handler, args, target, power, maxpower, powerMultiplier))
+		{
+			switch (target->getPowerType())
```
```
src/server/scripts/Commands/cs_modify.cpp:1009:12:
 fatal error:
 13 enumeration values not handled in switch: 'POWER_COMBO_POINTS', 'POWER_RUNES', 'POWER_SOUL_SHARDS'... [-Wswitch]
                        switch (target->getPowerType())
                                ^
1 error generated.
```
https://github.com/TrinityCore/TrinityCore/blob/2d75d8071bb6d85987bdc5d88ed2271822fcbcfd/src/server/game/Miscellaneous/SharedDefines.h#L250-L274

There are 22 `Powers` listed in the enum in `SharedDefines.h` which means you need to find a way to tell this script to skip/ignore the 13 powers not handled by this switch. I don't know what that code should look like, but maybe some of the other developers will know.

Jildor replied:
Need add a default in switch and add tabs in  cases

tkrokli replied:
OK, I get what you mean that it needs a
```cpp
			default:
				break;
```
at the end, but not sure what you mean by `add tabs in cases`

Jildor replied:
Add tabulations in  all cases:
```diff
-    case X
+        case X
```

Sorry mobile :smile:

tkrokli replied:
OK, I see. Add correct indent to each case,

tkrokli replied:
BTW; changes have been made, but they are all below this comment section
(and hence they don't get hidden by the update).

Jildor commented:
Add tabulations in  all cases:
```diff
-    case X
+        case X
```

Sorry mobile :smile:

Commit:
Update cs_modify.cpp

- add missing `default: break;` at end of the switch
- set proper indent (codestyle)

tkrokli commented:
OK - so far, so good. The C++ part compiles without errors.
Only the `sql/base/auth_database.sql` part left to fix to have travis-ci show up green.

Commit:
Update auth_database.sql

- add rbac_linked_permissions (198,867)
- add rbac_permissions (867, 'Command: modify power') after (854,'Command: .debug conversation')
- add updates ('2017_11_16_20833_auth.sql','42F7D718DBC00A0F5585BDCE66D1794BBD8267C5','RELEASED','2017-11-16 12:34:56',0)
  (hash from online SHA1 generator, date related to file name, time chosen randomly)

tkrokli commented:
@Hadifaren : I hope you take notes on my changes so you can use that knowledge later. 😃

Hadifaren commented:
@tkrokli of course your comments are so helpful ^_^

Hadifaren commented:
of course your comments always are helpful ^_^

Seyden commented:
A player can have multiple powers.. what you do then? What if i want to change the power that i got after i used a spell? With your change, it only uses my current power.

Also the power multiplier isn't always the same for each power type.

tkrokli commented:
@Seyden : Looks like your question is answered in the closing commit (reading the added content & commit message). 😄
